grakn-graph/src/main/java/ai/grakn/graph/internal/concept/TypeImpl.java

```diff
@@ -348,6 +348,35 @@ public T plays(Role role) {
     }
 
     /**
+     * This is a temporary patch to prevent accidentally disconnecting implicit {@link RelationType}s from their
+     * {@link RelationEdge}s. This Disconnection happens because {@link RelationType#instances()} depends on the
+     * presence of a direct {@link Schema.EdgeLabel#PLAYS} edge between the {@link Type} and the implicit {@link RelationType}.
+     *
+     * When changing the super you may accidentally cause this disconnection. So we prevent it here.
+     *
+     */
+    //TODO: Remove this when traversing to the instances of an implicit Relation Type is no longer done via plays edges
+    @Override
+    boolean changingSuperAllowed(T oldSuperType, T newSuperType){
+        boolean changingSuperAllowed = super.changingSuperAllowed(oldSuperType, newSuperType);
+        if(changingSuperAllowed && oldSuperType != null && !Schema.MetaSchema.isMetaLabel(oldSuperType.getLabel())) {
+            //noinspection unchecked
+            Set<Role> superPlays = TypeImpl.from(oldSuperType).directPlays().keySet();
+
+            superPlays.removeAll(directPlays().keySet());
+
+            //It is possible to be disconnecting from a role which is no longer in use but checking this will take too long
+            //So we assume the role is in sure and throw if that is the case
+            if(!superPlays.isEmpty() && instancesDirect().findAny().isPresent()){
+                throw GraphOperationException.changingSuperWillDisconnectRole(oldSuperType, newSuperType, superPlays.iterator().next());
+            }
+
+            return true;
+        }
+        return changingSuperAllowed;
+    }
```

Comment:
I think this is using methods like `directInstances` and `directPlays` when it shouldn't...

For example:
```
boy sub man;
man sub person;
person plays hungry;

$x isa boy;
(hungry: $x);
```

If we then go `boy sub rubiks-cube`, then the change should not be allowed because a `rubiks-cube` should not be hungry. We won't catch it because we only check `directPlays`.

A similar problem occurs if we go `man sub rubiks-cube`, because we only check `directInstances`.